Source/core/Sync.h

```diff
@@ -181,6 +182,44 @@ namespace Core {
     // For Windows platform compatibility
     typedef CountingSemaphore Semaphore;
 
+    // ===========================================================================
+    // class SharedSemaphore
+    // ===========================================================================
+
+    class EXTERNAL SharedSemaphore {
+    public:
+        SharedSemaphore() = delete;
+        SharedSemaphore(const Semaphore&) = delete;
+        SharedSemaphore& operator=(const SharedSemaphore&) = delete;
+
+#ifdef __WINDOWS__
+        SharedSemaphore(const TCHAR name[], const uint32_t initValue, const uint32_t maxValue);
```

Comment:
This could also work on Linux? Right, with the sem_open/sem_close? So why limit this to windows only ?